src/components/MAPI/organizations/utils.ts

```diff
@@ -64,3 +72,101 @@ export function computeClusterCountersForOrganizations(clusters?: Cluster[]) {
     return acc;
   }, {});
 }
+
+export async function fetchClusterListForOrganizations(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  cache: Cache,
+  provider: PropertiesOf<typeof Providers>,
+  organizations: Record<string, IOrganization>
+): Promise<ClusterList> {
```

Comment:
This is the function for fetching cluster list for organizations. First we perform a `SelfSubjectAccessReview` to determine if the user has permissions to list clusters at the cluster scope. If they do, we fetch the cluster list as we do before. If they do not, then we iterate through the namespaces they do have access to, and the cluster list from each namespace before aggregating.

The reason for performing the access check first and performing a request at the cluster scope if the user has permissions, is for performance reasons. For admin users with access to everything, the number of organizations they belong to would likely be extensive. During testing on `gauss`, it was taking a significantly longer time to iterate through all the organizations.

Reply:
:+1: